File: Medium/236. Lowest Common Ancestor of a Binary Tree/RecursiveSolution.java
class Solution {
    private TreeNode result = null;

    public TreeNode lowestCommonAncestor(TreeNode root, TreeNode p, TreeNode q) {
        path(root, p, q);
        return this.result;
    }

    private boolean path(TreeNode node, TreeNode p, TreeNode q) {
        if (node == null)
            return false;

        boolean cur = node == p || node == q;
        boolean left = path(node.left, p, q);
        boolean right = path(node.right, p, q);

        if ((cur && left) || (cur && right) || (left && right))
            this.result = node;

        return cur || left || right;
    }
}

File: 1. Easy/101. Symmetric Tree/recursive-solution.js
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isSymmetric = function (root) {
  return check(root.left, root.right);

  function check(left, right) {
    if (left == null && right == null) return true;
    if (left == null || right == null) return false;

    return (
      left.val == right.val &&
      check(left.left, right.right) &&
      check(left.right, right.left)
    );
  }
};
